### PR TITLE
Propagate indexes in DataArray binary operations.

### PR DESCRIPTION
**Pull request by dcherian**
Works by propagating indexes in DataArray._replace.

xref #2227. Tests pass!

 - [x] Tests added
 - [x] Passes `black . && mypy . && flake8`

### CODE REVIEW

File: xarray/core/indexes.py [hunk 1]
```diff
@@ -35,6 +35,9 @@ def __contains__(self, key):
     def __getitem__(self, key):
         return self._indexes[key]
 
+    def __delitem__(self, key):
```
Comment (dcherian) [inline xarray/core/indexes.py]: needed for the groupby change above.

File: xarray/core/dataarray.py [hunk 2]
```diff
@@ -386,14 +386,15 @@ def _replace(
...
+        indexes=None,
     ) -> "DataArray":
         if variable is None:
             variable = self.variable
         if coords is None:
             coords = self._coords
         if name is _default:
             name = self.name
-        return type(self)(variable, coords, name=name, fastpath=True)
+        return type(self)(variable, coords, name=name, fastpath=True, indexes=indexes)
```
Comment (max-sixty) [inline xarray/core/dataarray.py]: Do we need to do this in `.copy` (which I think calls this method)?

Comment (dcherian) [reply xarray/core/dataarray.py]: I can do that in a followup.

Comment (max-sixty) [reply xarray/core/dataarray.py]: Great, yes only if needed, 👍  re small PRs in

**Pull request merged**